react-search/my-app/src/SearchableList.tsx

```diff
@@ -0,0 +1,51 @@
+import { Dispatch, SetStateAction, useState } from 'react';
+
+type Props = {
+  list: string[];
+};
+type SearchBar = {
+  search: string;
+  setSearch: Dispatch<SetStateAction<string>>;
+};
+type ListItems = {
+  items: string[];
+  search: string;
+};
+
+export function SearchableList({ list }: Props) {
+  const [search, setSearch] = useState('');
+
+  console.log(search);
+
+  return (
+    <div>
+      <SearchBar search={search} setSearch={setSearch} />
+      <ListItems items={list} search={search} />
+    </div>
+  );
+}
+
+function SearchBar({ search, setSearch }: SearchBar) {
+  return (
+    <input
+      type="text"
+      value={search}
+      onChange={(e) => setSearch(e.target.value)}
+    />
+  );
+}
+
+function ListItems({ items, search }: ListItems) {
```

Comment:
You could also filter the items up in SearchableList and pass just the ones you want rendered into this component.